src/indexing_slicing.rs

```diff
@@ -0,0 +1,54 @@
+use rustc::lint::*;
+use rustc::middle::const_eval::EvalHint::ExprTypeChecked;
+use rustc::middle::const_eval::{eval_const_expr_partial, ConstVal};
+use rustc::middle::ty::TyArray;
+use rustc_front::hir::*;
+use utils::span_lint;
+
+/// **What it does:** Check for usage of indexing or slicing.
+///
+/// **Why is this bad?** This may panic at runtime.
```

Comment:
You might want to say that it’s usually not bad except for some specific application and point out that there are non-panicking alternatives.